### PR TITLE
Move engine startup into helper function

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

This removes a _bit_ of duplicated code, and moves the functions so that the order that they are included in the files flow with the expected call order. Since many of our syncing protocols flow from step 1 -> step 2 -> ..., I feel like the code should also read from step 1 -> step 2 -> .... as much as possible.

## How this works

Super minor refactor - pulling stuff out of #2297

## How this was tested

CI

### CODE REVIEW

File: snow/engine/snowman/bootstrap/bootstrapper.go [hunk 1]
```diff
@@ -149,7 +149,40 @@ func (b *bootstrapper) Start(ctx context.Context, startReqID uint32) error {
 	b.startingHeight = lastAccepted.Height()
 	b.Config.SharedCfg.RequestID = startReqID
 
-	if !b.StartupTracker.ShouldStart() {
+	return b.startBootstrappingOnceIfSufficientlyConnected(ctx)
```
Comment (abi87) [inline snow/engine/snowman/bootstrap/bootstrapper.go:152]: `startBootstrappingOnceIfSufficientlyConnected` is pretty clear but I think their are not really idomatic in go. How about we rename to something like "tryStart" and we explain in a comment we're looking for enough connected stake?

File: snow/engine/snowman/syncer/state_syncer.go [hunk 2]
```diff
@@ -112,6 +112,51 @@ func (ss *stateSyncer) Context() *snow.ConsensusContext {
...
+		Type:  p2p.EngineType_ENGINE_TYPE_SNOWMAN,
+		State: snow.StateSyncing,
+	})
+	if err := ss.VM.SetState(ctx, snow.StateSyncing); err != nil {
+		return fmt.Errorf("failed to notify VM that state syncing has started: %w", err)
+	}
+
+	ss.requestID = startReqID
+
+	return ss.startSyncingOnceIfSufficientlyConnected(ctx)
```
Comment (abi87) [inline snow/engine/snowman/syncer/state_syncer.go:128]: same as above, I'd suggest `tryStart`

**Review by abi87 — APPROVED** [review]
just a nit around naming

File: snow/engine/snowman/bootstrap/bootstrapper.go (hunk 1 above)
Comment (StephenButtolph) [reply snow/engine/snowman/bootstrap/bootstrapper.go:152]: ok

**Review by dhrubabasu — APPROVED** [review]

**Pull request merged**